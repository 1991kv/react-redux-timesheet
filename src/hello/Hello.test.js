import React from 'react';
import Hello from './Hello';
import renderer from 'react-test-renderer';

describe('Hello World:', function () {


    it('should render with default text', function () {

        const component = renderer.create(
                <Hello/>
        );

        let stringVal = JSON.stringify(component);
        expect(stringVal).toMatch(/Howdy/);
        expect(stringVal).toMatch(/Partner/);

    });

    it('should render with our props', function () {

        const component = renderer.create(
                <Hello friend="Fred"/>
        );

        let stringVal = JSON.stringify(component);
        expect(stringVal).toMatch(/Howdy/);
        expect(stringVal).toMatch(/Fred/);

    });
    
    it('should render to match the snapshot', function () {

        const component = renderer.create(
            <Hello friend="Luke"/>
        );

        const json = component.toJSON();
        expect(json).toMatchSnapshot();
    });    


});



