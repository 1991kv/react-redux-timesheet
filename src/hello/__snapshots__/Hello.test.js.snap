exports[`Hello World: should render to match the snapshot 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="hello">
    <h1>
        Howdy!!
    </h1>
    <h2>
        Luke
    </h2>
    <p>
        Congratulations!  You have created your first React component!
    </p>
</div>,
  "nodes": Array [
    <div
      className="hello">
      <h1>
            Howdy!!
      </h1>
      <h2>
            Luke
      </h2>
      <p>
            Congratulations!  You have created your first React component!
      </p>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <Hello
        friend="Luke" />,
      "_debugID": 7,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Hello {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "friend": "Luke",
        },
        "refs": Object {},
        "state": Object {
          "greeting": "Howdy!!",
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 4,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className="hello">
          <h1>
                    Howdy!!
          </h1>
          <h2>
                    Luke
          </h2>
          <p>
                    Congratulations!  You have created your first React component!
          </p>
</div>,
        "_debugID": 8,
        "_renderedOutput": <div
          className="hello">
          <h1>
                    Howdy!!
          </h1>
          <h2>
                    Luke
          </h2>
          <p>
                    Congratulations!  You have created your first React component!
          </p>
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Hello
    friend="Luke" />,
}
`;
